src/nvim/state.c

```diff
@@ -62,6 +63,16 @@ void state_enter(VimState *s)
       }
       // Flush screen updates before blocking
       ui_flush();
+
+      // Skip everything before VimEnter event, like calling prompt() in init.vim
+      if (time_fd != NULL && get_vim_var_nr(VV_VIM_DID_ENTER)) {
```

Comment:
`v:vim_did_enter` is set before `VimEnter` event, so everything from `VimEnter autocommands` can still be cut short if something like `prompt()` is called inside a `VimEnter` autocommand.